nsqd/diskqueue.go

```diff
@@ -261,6 +261,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
 		return nil, err
 	}
 
+	if msgSize < minMsgSize {
```

Comment:
`DiskQueue` doesn't (and arguably shouldn't) know about "messages".  I think that `DiskQueue` should test for `msgSize > 0`, not `minMsgSize`.  The receiver of the read channel should enforce that the size is appropriate for a message, i.e. at least `minMsgSize`, so we can create an additional check there.